### PR TITLE
Chart store features & bug fixes

### PR DESCRIPTION
**Pull request by milindk8**
# Description

Added some features & bug fixes in the chart store

1. Search: Update no result empty state
2. Multi-chart deploy right panel interactions
           - Hide [+ Add Chart] when the chart list is visible
           - Select charts → [+ Add Chart]
           - Add the [Remove All] button
           - [BUG] Advanced options should be shown when the user clicks on [+ Add Chart]    
           - Multi-chart deploy: Add back to Advanced options page (independent)

Fixes - https://github.com/devtron-labs/devtron/issues/2221

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

Screenshot 2022-09-06 at 3 59 40 PM

Screenshot 2022-09-06 at 4 01 26 PM
ercontent.com/98738644/188613791-d7b3e818-f973-4156-8c45-ba5208e44281.png">

# How Has This Been Tested?
All the testing has been done manually

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/charts/list/list.scss [hunk 1]
```diff
@@ -161,7 +161,7 @@
 }
 
 .chart-count {
-    padding: 16px 0;
+    padding:0 0 16px 0;
```
Comment (sohel-rp) [inline src/components/charts/list/list.scss:164]: Just say - 0 0 16px

Comment (milindk8) [reply src/components/charts/list/list.scss]: Done

File: src/components/charts/useChartGroup.ts [hunk 2]
```diff
@@ -324,12 +324,13 @@ export default function useChartGroup(chartGroupId = null): ChartGroupExports {
     }
 
     function removeChart(index: number): void {
-        const tempCharts = [...state.charts]
-        tempCharts.splice(index, 1)
+        let tempCharts = [...state.charts]
+        index === -1 ? tempCharts = [] : tempCharts.splice(index, 1)
```
Comment (sohel-rp) [inline src/components/charts/useChartGroup.ts:328]: Use if statement

File: src/components/charts/list/DiscoverCharts.tsx [hunk 3]
```diff
@@ -364,43 +391,42 @@ function DiscoverChartList() {
...
+                                                                                state.selectedInstances[chart.id]
+                                                                                    ?.length
+                                                                            }
+                                                                            showCheckBoxOnHoverOnly={
+                                                                                state.charts.length === 0
+                                                                            }
+                                                                            addChart={addChart}
+                                                                            showDescription={!isGrid}
+                                                                            subtractChart={subtractChart}
+                                                                            onClick={(chartId) =>
```
Comment (sohel-rp) [inline src/components/charts/list/DiscoverCharts.tsx:414]: extract this function

File: src/components/charts/list/DiscoverCharts.tsx [hunk 4]
```diff
@@ -276,6 +286,23 @@ function DiscoverChartList() {
         history.push(url)
     }
 
+    const viewChartRepo = () => {
+        return (
+            <div className="pt-10 pb-10 pl-12 pr-12 br-4 bw-1 bcv-1 w-100 ev-2 flexbox mxw-300">
```
Comment (sohel-rp) [inline src/components/charts/list/DiscoverCharts.tsx:291]: Use InfoColourBar component

File: src/components/common/emptyState/ChartEmptyState.tsx [hunk 5]
```diff
@@ -8,9 +8,10 @@ interface EmptyChartType {
     onClickViewChartButton: () => void;
     buttonText?: string;
     heightToDeduct?: number;
+    addComponent?: JSX.Element;
 }
 
-function ChartEmptyState({ title, subTitle, onClickViewChartButton, buttonText, heightToDeduct }: EmptyChartType) {
+function ChartEmptyState({ title, subTitle, onClickViewChartButton, buttonText, heightToDeduct, addComponent }: EmptyChartType) {
```
Comment (sohel-rp) [inline src/components/common/emptyState/ChartEmptyState.tsx:14]: Take additional custom components as children.

File: src/components/charts/MultiChartSummary.tsx [hunk 6]
```diff
@@ -30,33 +31,48 @@ const MultiChartSummary: React.FC<MultiChartSummaryProps> = ({
...
-                        <Forward className="rotate anchor" style={{ ['--rotateBy' as any]: '180deg' }} />
-                        <div className="anchor ml-18">Select Charts</div>
-                        <span />
-                    </div>
-
+                    {name && (
+                        <div className="flex column left border-bottom mb-20">
+                            <span className="flex flex-justify w-100 fs-14 cn-9">
+                                Group name
+                                <Pencil className="pointer" onClick={(e) => setChartDetailsUpdate(true)} />
```
Comment (vivek-devtron) [inline src/components/charts/MultiChartSummary.tsx:47]: We create separate function for {(e) => setChartDetailsUpdate(true)} and avoid inline function

File: src/css/base.scss [hunk 7]
```diff
@@ -1869,6 +1869,10 @@ button.anchor {
   min-width: 48px;
 }
 
+.mxw-300 {
```
Comment (vivek-devtron) [inline src/css/base.scss:1872]: make it dc__mxw-300

Comment (shivani170) [inline src/css/base.scss:1872]: all the base scss will initialize with prefix `dc__`

File: src/components/charts/list/list.scss [hunk 8]
```diff
@@ -161,7 +161,7 @@
 }
 
 .chart-count {
-    padding: 16px 0;
+    padding:0 0 16px;
```
Comment (shivani170) [inline src/components/charts/list/list.scss:164]: Can we use direct formulae if possible

File: src/components/charts/useChartGroup.ts (hunk 2 above)
Comment (milindk8) [reply src/components/charts/useChartGroup.ts]: Done

File: src/components/common/emptyState/ChartEmptyState.tsx (hunk 5 above)
Comment (milindk8) [reply src/components/common/emptyState/ChartEmptyState.tsx]: Done

File: src/components/charts/list/list.scss (hunk 8 above)
Comment (milindk8) [reply src/components/charts/list/list.scss]: Done

File: src/css/base.scss (hunk 7 above)
Comment (milindk8) [reply src/css/base.scss]: done

File: src/components/charts/MultiChartSummary.tsx (hunk 6 above)
Comment (milindk8) [reply src/components/charts/MultiChartSummary.tsx]: done

File: src/components/charts/MultiChartSummary.tsx [hunk 9]
```diff
@@ -7,6 +7,7 @@ import { ReactComponent as Trash } from '../../assets/icons/ic-delete.svg'
 import { ReactComponent as Warning } from '../../assets/icons/ic-warning.svg'
 import { ReactComponent as EmptyFolder } from '../../assets/icons/img-folder-empty.svg'
 import DropDownFilled from '../../assets/icons/ic-dropdown-filled.svg'
+import { ReactComponent as Add } from '../../assets/icons/ic-add.svg'
 
 interface MultiChartSummaryProps extends ChartSummaryHelpers {
```
Comment (vivek-devtron) [inline src/components/charts/MultiChartSummary.tsx:12]: Plesae move this to props

File: src/components/charts/list/DiscoverCharts.tsx [hunk 10]
```diff
@@ -42,6 +42,8 @@ import SavedValuesList from '../SavedValues/SavedValuesList'
 import ChartValues from '../chartValues/ChartValues'
 import { ReactComponent as Next } from '../../../assets/icons/ic-arrow-forward.svg'
 import NoGitOpsConfiguredWarning from '../../workflowEditor/NoGitOpsConfiguredWarning'
+import { ReactComponent as Help } from '../../../assets/icons/ic-help.svg'
+import { ReactComponent as BackIcon } from '../../../assets/icons/ic-back.svg'
 
 interface EmptyCharts {
```
Comment (vivek-devtron) [inline src/components/charts/list/DiscoverCharts.tsx:48]: move this to types

File: src/components/charts/MultiChartSummary.tsx (hunk 9 above)
Comment (milindk8) [reply src/components/charts/MultiChartSummary.tsx]: Done

File: src/components/charts/list/DiscoverCharts.tsx (hunk 10 above)
Comment (milindk8) [reply src/components/charts/list/DiscoverCharts.tsx]: Done

File: src/components/charts/list/DiscoverCharts.tsx [hunk 11]
```diff
@@ -276,6 +274,25 @@ function DiscoverChartList() {
         history.push(url)
     }
 
+    const ViewChartRepo = () => {
+        return (
+            <ChartEmptyState onClickViewChartButton={clearSearch}>
+                <div className="pt-10 pb-10 pl-12 pr-12 br-4 bw-1 bcv-1 w-100 ev-2 flexbox dc__mxw-300">
```
Comment (shivani170) [inline src/components/charts/list/DiscoverCharts.tsx:280]: Info colour bar?

Comment (milindk8) [reply src/components/charts/list/DiscoverCharts.tsx]: Done

File: src/components/charts/list/DiscoverCharts.tsx (hunk 4 above)
Comment (milindk8) [reply src/components/charts/list/DiscoverCharts.tsx]: Done

File: src/components/charts/list/DiscoverCharts.tsx (hunk 3 above)
Comment (milindk8) [reply src/components/charts/list/DiscoverCharts.tsx:386]: After extracting, for every selection api for icon hits which reload the icon

File: src/components/charts/list/DiscoverCharts.tsx [hunk 12]
```diff
@@ -364,43 +363,48 @@ function DiscoverChartList() {
...
-                                                            subTitle={`We couldn't find any matching results`}
-                                                            styles={{ height: '300px', justifyContent: 'center' }}
-                                                        />
+                                                        <ChartEmptyState onClickViewChartButton={clearSearch}>
+                                                            <InfoColourBar
+                                                                message={'Can’t find what you’re looking for?'}
+                                                                classname={'br-4 bw-1 bcv-1 ev-2 dc__mxw-300 bcv-1 fs-12 pl-12 pr-12'}
+                                                                Icon={Help}
+                                                                iconClass={'fcv-5 h-20'}
+                                                                linkText={'Try refetching connected chart repos or connect a chart repository'}
```
Comment (sohel-rp) [inline src/components/charts/list/DiscoverCharts.tsx:404]: Directly pass it as string. No need to wrap around {}

File: src/components/charts/useChartGroup.ts [hunk 13]
```diff
@@ -323,13 +323,30 @@ export default function useChartGroup(chartGroupId = null): ChartGroupExports {
         setState((state) => ({ ...state, charts: tempCharts }))
     }
 
-    function removeChart(index: number): void {
-        const tempCharts = [...state.charts]
-        tempCharts.splice(index, 1)
+    function removeChart(index: number, removeAll?: boolean): void {
+        let tempCharts = [...state.charts]
+        if (removeAll) {
+            tempCharts = []
```
Comment (sohel-rp) [inline src/components/charts/useChartGroup.ts:329]: Replace it with `tempCharts.length = 0`

File: src/components/common/infocolourBar/InfoColourbar.tsx [hunk 14]
```diff
@@ -7,17 +7,29 @@ interface InfoColourBarType {
...
+function InfoColourBar({ message, classname, Icon, iconClass, renderActionButton, redirectToLink, linkText }: InfoColourBarType) {
     return (
         <div className="info-bar-container">
             <div className={`${classname} info_text flex content-space pt-10 pb-10 pl-16 pr-16 br-4 top fs-13 fw-4`}>
                 <span className="flex top">
                     <div className="icon-dim-20 mr-10">
                         <Icon className={`${iconClass} icon-dim-20 mr-8`} />
                     </div>
-                    {message}
+                    <span>
```
Comment (sohel-rp) [inline src/components/common/infocolourBar/InfoColourbar.tsx:22]: Change this to div and use span for the message, remove br & wrap a tag in another div.

File: src/components/charts/list/DiscoverCharts.tsx (hunk 12 above)
Comment (milindk8) [reply src/components/charts/list/DiscoverCharts.tsx]: done

File: src/components/charts/useChartGroup.ts (hunk 13 above)
Comment (milindk8) [reply src/components/charts/useChartGroup.ts]: done

File: src/components/common/infocolourBar/InfoColourbar.tsx (hunk 14 above)
Comment (milindk8) [reply src/components/common/infocolourBar/InfoColourbar.tsx]: Done

**Review by vivek-devtron — APPROVED** [review]

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**